src/cljs/vanilla/widgets/configure_widget.cljs

```diff
@@ -0,0 +1,212 @@
+(ns vanilla.widgets.configure-widget
+  (:require [reagent.core :as r]
+            [re-frame.core :as rf]
+            [day8.re-frame.tracing :refer-macros [fn-traced]]
+            [cljsjs.react-color]
+            [vanilla.update-layout :as u]))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;
+;
+; helpers
+;
+
+(def def-color {:r 150 :g 150 :b 150 :a 1})
+
+(defn- f [n w]
+  (->> w
+     (filter #(= (:key %) n))
+     first)) ; returns a map
+
+(defn- notf [n w]
+  (->> w
+       (filter #(not= (:key %) n)))) ; a seq of maps
+
+(defn- modify-the-widget [m banner-color title-color title]
+  (-> m
+      (assoc-in [:options :viz/title] title)
+      (assoc-in [:options :viz/banner-text-color] title-color)
+      (assoc-in [:options :viz/banner-color] banner-color)))
+
+(defn- get-part [id widget path default]
+  (if (and (not= id "")
+           (not (nil? id)))
+    (get-in widget path default)
+    default))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;
+;
+; event handlers and subscriptions
+;
+
+(rf/reg-event-db
+  :configure-widget
+  (fn-traced
+    [db [_ widget-id]]
+    (let [widget (f widget-id (:widgets db))]
+      (assoc db
+        :configure-widget widget-id
+        :chosen-widget (if (= widget-id "") nil widget)
+        :chosen-title (get-part widget-id widget [:options :viz/title] "")
+        :chosen-banner-color (get-part widget-id widget [:options :viz/banner-color] def-color)
+        :chosen-title-color (get-part widget-id widget [:options :viz/banner-text-color] def-color)))))
+
+
+
+(rf/reg-event-db
+  :chosen-title
+  (fn-traced
+    [db [_ val]]
+    (assoc db :chosen-title val)))
+
+
+
+(rf/reg-event-db
+  :chosen-title-color
+  (fn-traced
+    [db [_ val]]
+    (assoc db :chosen-title-color val)))
+
+
+(rf/reg-event-db
+  :chosen-banner-color
+  (fn-traced
+    [db [_ val]]
+    (assoc db :chosen-banner-color val)))
+
+
+
+(rf/reg-event-db
+  :update-widget-config
+  (fn-traced
+    [db [_ widget-id]]
+    (let [orig-widget (f widget-id (:widgets db))
+          mod-widget (modify-the-widget orig-widget
+                       (:chosen-banner-color db)
+                       (:chosen-title-color db)
+                       (:chosen-title db))
+          new-widgets (into [] (conj (notf widget-id (:widgets db)) mod-widget))]
+      (u/save-layout new-widgets)
```

Comment:
So has this call to u/save-layout effectively replaced the update-widget POST that was removed from basic widget?  This is definitely cleaner, just making sure I understand.

Reply:
yes. I initially tried using the older call, but I wasn't sure how well it had been tested, and I knew that (save-layout) was pretty complete an solid, so it seemed the better part of valor to just use it. We may want to revisit that decision later, if we ever reach a point where user have dozens of widgets and we don't want to just "save everything"